src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,56 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.InvalidRegistrationDataException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int LOGIN_UPPER_BOUND = 10;
+    private static final int PASSWORD_UPPER_BOUND = 16;
+    private static final int PASSWORD_LOWER_BOUND = 6;
+    private static final int AGE_UPPER_BOUND = 130;
+    private static final int AGE_LOWER_BOUND = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidRegistrationDataException("Null user value");
+        }
+        if (storageDao.get(user.getLogin()) == null) {
+            isValidData(user);
+            return storageDao.add(user);
+        }
+        throw new InvalidRegistrationDataException(
+                "Can't register a user with existing login " + user.getLogin());
+    }
+
+    private void isValidData(User user) {
```

Comment:
this method does not return anything to be called this way
```suggestion
    private void checkUser(User user) {
```